apps/framework-cli/src/cli/local_webserver.rs

```diff
@@ -1290,30 +1289,109 @@ fn handle_listener_err(port: u16, e: std::io::Error) -> ! {
         _ => panic!("Failed to listen to port {}: {:?}", port, e),
     }
 }
-async fn shutdown(settings: &Settings, project: &Project, graceful: GracefulShutdown) -> ! {
+async fn shutdown(
+    settings: &Settings,
+    project: &Project,
+    graceful: GracefulShutdown,
+    redis_client: Arc<Mutex<RedisClient>>,
+) -> ! {
+    // First, initiate the graceful shutdown of HTTP connections
+    let shutdown_future = graceful.shutdown();
+
+    // Wait for connections to close with a timeout
     tokio::select! {
-            _ = graceful.shutdown() => {
-                info!("all connections gracefully closed");
-            },
-            _ = tokio::time::sleep(std::time::Duration::from_secs(10)) => {
-                warn!("timed out wait for all connections to close");
+        _ = shutdown_future => {
+            info!("all connections gracefully closed");
+        },
+        _ = tokio::time::sleep(std::time::Duration::from_secs(10)) => {
+            warn!("timed out wait for all connections to close");
         }
     }
 
+    // Ensure any Redis-using tasks are stopped
+    info!("Stopping any tasks that might be using Redis");
+    tokio::time::sleep(std::time::Duration::from_millis(100)).await;
```

Comment:
why do we have another sleep here having waiting 10s above worst case / if shutdown was graceful?

Reply:
That was left over from an earlier iterations.  I can be safely removed.  Will test before next update to this PR.